debezium-core/src/main/java/io/debezium/pipeline/source/snapshot/incremental/AbstractIncrementalSnapshotChangeEventSource.java

```diff
@@ -356,10 +358,25 @@ private boolean isTableInvalid(P partition, OffsetContext offsetContext) {
         final TableId currentTableId = (TableId) context.currentDataCollectionId().getId();
         currentTable = databaseSchema.tableFor(currentTableId);
         if (currentTable == null) {
-            LOGGER.warn("Schema not found for table '{}', known tables {}", currentTableId, databaseSchema.tableIds());
-            notificationService.incrementalSnapshotNotificationService().notifyTableScanCompleted(context, partition, offsetContext, totalRowsScanned, UNKNOWN_SCHEMA);
-            nextDataCollection(partition, offsetContext);
-            return true;
+            LOGGER.info("Schema not found for table '{}', known tables {}. Will attempt to retrieve this schema", currentTableId, databaseSchema.tableIds());
+            try {
+                retrieveAndRefreshSchema(currentTableId, partition, offsetContext);
+                currentTable = databaseSchema.tableFor(currentTableId);
+                if (currentTable == null) {
+                    LOGGER.warn("Schema retrieval failed to populate the schema as expected for {}", currentTableId);
+                    notificationService.incrementalSnapshotNotificationService().notifyTableScanCompleted(context, partition, offsetContext, totalRowsScanned,
+                            UNKNOWN_SCHEMA);
+                    nextDataCollection(partition, offsetContext);
+                    return true;
```

Comment:
Maybe you can put these into a method with name `warnAndSkip(currentTableId)`. WDYT?

Reply:
@mfvitale  in latest commit I moved warnAndSkip into its own method and I'm using it for the isTableInvalid warnings.  Hopefully this aligns with your suggestion above?